src/model/auto-prompt-config.js

```diff
@@ -19,10 +19,64 @@
  */
 export class AutoPromptConfig {
   /**
-   * @param {number} maxImpressionsPerWeek
+   * @param {number|undefined} maxImpressionsPerWeek
    */
-  constructor(maxImpressionsPerWeek) {
-    /** @const {number} */
+  constructor(
+    maxImpressionsPerWeek,
+    dismissalDelaySeconds,
+    backoffSeconds,
+    maxDismissalsPerWeek,
+    maxDismissalsResultingHideSeconds
+  ) {
+    /** @const {number|undefined} */
     this.maxImpressionsPerWeek = maxImpressionsPerWeek;
+
+    /** @const {!ClientDisplayTrigger} */
+    this.clientDisplayTrigger = new ClientDisplayTrigger(dismissalDelaySeconds);
+
+    /** @const {!ExplicitDismissalConfig} */
+    this.explicitDismissalConfig = new ExplicitDismissalConfig(
+      backoffSeconds,
+      maxDismissalsPerWeek,
+      maxDismissalsResultingHideSeconds
+    );
+  }
+}
+
+/**
+ *
+ */
+export class ClientDisplayTrigger {
+  /**
+   * @param {number|undefined} dismissalDelaySeconds
+   */
+  constructor(dismissalDelaySeconds) {
+    /** @const {number|undefined} */
+    this.dismissalDelaySeconds = dismissalDelaySeconds;
+  }
+}
+
+/**
+ *
```

Comment:
Could you add a description for this class?

Reply:
Done.